7828 name only rewrite (#2)
jrudolf/ERCs: ERCS/erc-7828.md

---

## Abstract

> This proposal builds off of [ERC-X] (Interoperable Addresses) and [ERC-7785] (on-chain configs) to provide a standard and human-readable format for chain-specific addresses which provides:

Reviewer: ```suggestion
This proposal builds off of [ERC-7930] (Interoperable Addresses) and [ERC-7785] (on-chain configs) to provide a standard and human-readable format for chain-specific addresses which provides:
```
we need to link it too after merge.

A few examples below:
```
Mainnet
- 0x12345...6789@eip155:1#FFFFFFFF
- 0x12345...6789@eth.short#FFFFFFFF
- alice.eth@eth.short#FFFFFFFF

Testnet (Sepolia)
- 0x12345...6789@eip155:11155111#00000000
- alice.testeth@sep.short#00000000

Rollup
- 0x12345...6789@eip155:4270#AAAAAAAA
- 0x12345...6789@arb-nova.short#AAAAAAAA
- alice.eth@arb-nova.short#AAAAAAAA

Assuming ERC-7785 starts listing chains at l2.eth
- alice.eth@arb-nova.l2.eth#AAAAAAAA

My ENS name is registered on rollup1 (via ENSIP-10 & ENSIP-16), but I want to receive funds on rollup2
- alice.rollup1.eth@rollup2.short#BBBBBBBB

Non-evm chain
bc1..23@bip122:000000000019d6689c085ae165831e93#CCCCCCCC
alice.eth@bip122:000000000019d6689c085ae165831e93#CCCCCCCC

Assuming the shortcode list adds a few other CAIP namespaces
alice.eth@btc.short
```

### Checksum
Addreses must be serialized to the `ChainType, ChainReferenceLength, ChainReference, AddressLength, Address` format proposed in [ERC-X] as Interoperable Addresses v1, and the 4-byte checksum MUST be displayed as part of the address as described in the syntax above.

> This has the property of the checksum changing if any part of the address or chainid is modified, making dangerous manual modification of addresses an error case for wallet users, and also mitigates collisions where the same name is registered to separate addresses on 

Reviewer 2: I would rewrite this for clarity
```suggestion
This ensures that the checksum changes if any part of the address or chain ID is modified, turning dangerous manual edits into detectable errors for wallet users. It also helps prevent collisions where the same name is registered to different addresses on separate chains.
```

Author: thanks! implemented it, but also removed the part about name collisions given the centralized name registry registry should avoid it

### The Name Resolver Registry

> We propose for this ERC to strive to reach Final status instead of becoming a Living document, and maintain a separate dictionary for the different name resolving protocols or methods, similar to ethereum-lists/chains.

Reviewer 2: ```suggestion
We propose for this ERC to strive to reach Final status instead of becoming a living document, and maintain a separate dictionary for the different name resolving protocols or methods, similar to ethereum-lists/chains.
```
